### PR TITLE
chore: migrates config types (core migration 3)

### PR DESCRIPTION
**Pull request by haverchuck**
*Description of changes:*
Depends on #1552 

Migrates config types from amplify_flutter to amplify_core.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_core/lib/src/category/amplify_auth_category.dart [hunk 1]
```diff
@@ -19,10 +19,10 @@ part of amplify_interface;
...
-  const AuthCategory();
-
-  static List<AuthPluginInterface> plugins = [];
+class AuthCategory extends AmplifyCategory<AuthPluginInterface> {
+  @override
+  @nonVirtual
+  Category get category => Category.auth;
 
   /// `Add plugin` method
   Future<void> addPlugin(AuthPluginInterface plugin) async {
```
Comment (dnys1) [inline packages/amplify_core/lib/src/category/amplify_auth_category.dart:28]: This is already defined on `AmplifyCategory`

File: packages/amplify_core/lib/src/types/auth/attribute/auth_next_update_attribute_step.dart [hunk 2]
```diff
@@ -30,8 +30,8 @@ class AuthNextUpdateAttributeStep extends AuthNextStep {
 
   // ignore: public_member_api_docs
   AuthNextUpdateAttributeStep(
-      {additionalInfo,
-      required codeDeliveryDetails,
+      {Map<dynamic, dynamic>? additionalInfo,
```
Comment (dnys1) [inline packages/amplify_core/lib/src/types/auth/attribute/auth_next_update_attribute_step.dart:33]: `Map<String, String>`

File: packages/amplify_core/lib/src/types/auth/sign_in/auth_next_sign_in_step.dart [hunk 3]
```diff
@@ -13,13 +13,15 @@
...
-import '../types/AuthNextStep.dart';
+import 'package:amplify_core/amplify_core.dart';
 
 class AuthNextSignInStep extends AuthNextStep {
   String signInStep;
-  AuthNextSignInStep(
-      {additionalInfo, codeDeliveryDetails, required this.signInStep})
-      : super(
+  AuthNextSignInStep({
+    Map<String, dynamic>? additionalInfo,
```
Comment (dnys1) [inline packages/amplify_core/lib/src/types/auth/sign_in/auth_next_sign_in_step.dart:21]: `Map<String, String>`

File: packages/amplify_core/lib/src/types/auth/sign_out/sign_out_request.dart [hunk 4]
```diff
@@ -13,7 +13,9 @@
  * permissions and limitations under the License.
  */
 
-import './SignOutOptions.dart';
+// ignore_for_file: implicit_dynamic_parameter
```
Comment (dnys1) [inline packages/amplify_core/lib/src/types/auth/sign_out/sign_out_request.dart:16]: Why though? I literally fixed all these in AD lol

**Review by dnys1 — COMMENTED** [review]
`ignore_for_file` should be throwing up red flags in your mind. All of them can be removed, they were fixed in AD which is why I said to bring over those files.

**Review by Jordan-Nelson — APPROVED** [review]

File: packages/amplify_core/pubspec.yaml [hunk 5]
```diff
@@ -8,13 +8,14 @@ environment:
   flutter: ">=1.17.0"
 
 dependencies:
+  aws_common: 0.1.0
```
Comment (dnys1) [inline packages/amplify_core/pubspec.yaml:11]: aws packages will follow semver so this should be `^0.1.0` throughout

File: packages/auth/amplify_auth_cognito/lib/method_channel_auth_cognito.dart [hunk 6]
```diff
@@ -25,7 +25,8 @@ const MethodChannel _channel =
     MethodChannel('com.amazonaws.amplify/auth_cognito');
 
 /// An implementation of [AmplifyAuthCognito] that uses method channels.
-class AmplifyAuthCognitoMethodChannel extends AmplifyAuthCognito {
+class AmplifyAuthCognitoMethodChannel extends AmplifyAuthCognito
+    implements PlatformInterface {
```
Comment (dnys1) [inline packages/auth/amplify_auth_cognito/lib/method_channel_auth_cognito.dart:29]: Remove

File: packages/aws_common/pubspec.yaml [hunk 7]
```diff
@@ -12,6 +12,7 @@ dependencies:
   collection: ^1.15.0
   http: ^0.13.4
   meta: ^1.7.0
+  uuid: ^3.0.1
```
Comment (dnys1) [inline packages/aws_common/pubspec.yaml:15]: I would think this would stay in amplify_core - what was the intention here?

Comment (haverchuck) [reply packages/aws_common/pubspec.yaml:15]: My thinking was that aws_common could have any generic utils that do not have dependencies on Flutter or other Amplify-related types.

Comment (dnys1) [reply packages/aws_common/pubspec.yaml:15]: Gotcha- sounds good thanks. I'll want to change the interface and add doc comments but I'll do that in a follow up PR

**Review by dnys1 — APPROVED** [review]

**Pull request merged**